### PR TITLE
Validate event properties

### PR DESCRIPTION
**Pull request by n3f**
Fixes https://github.com/Automattic/gold/issues/525

#### Changes proposed in this Pull Request

* Add event validation.
* Fixes flaky tests (github action didn't like caching strategy)
* Document/add an alternative testing strategy.

#### Testing instructions

* Setup with `wp-env` (e.g. `npm i && composer install && npx wp-env start`)
* `npm run test`

### CODE REVIEW

**Comment by n3f** [comment]
I guess the tests are flaky...

File: .github/workflows/php-phpunit.yml [hunk 1]
```diff
@@ -54,11 +54,11 @@ jobs:
           echo "WPENV_DIR=$(npx wp-env install-path)" >> $GITHUB_OUTPUT
 
       # setup the wp-env cache
-      - name: Cache wp-env
-        uses: actions/cache@v4
-        with:
-          path: ${{ steps.wp_env_dir.outputs.WPENV_DIR }}
-          key: ${{ runner.os }}-${{ hashFiles('**/.wp-env.json') }}
+#      - name: Cache wp-env
```
Comment (millerf) [inline .github/workflows/php-phpunit.yml:57]: @n3f Is that change necessary? If so, should it be removed?

Comment (n3f) [reply .github/workflows/php-phpunit.yml:57]: So the builds were failing with this in for some reason.  I'm not sure if it created a timeout or if the cache was bad or what.  I left the comment in because it seems like it should work (and it did work for several builds - then it broke for several). Since I removed it, all the builds have been working correctly.

File: src/inc/woocommerce-actions.php [hunk 2]
```diff
@@ -120,7 +126,7 @@ public static function login_failure( string $username, \WP_Error $error ) {
 				$failure_reason = null;
 		}
 		$properties = array(
-			'$user_id'      => $attempted_user->ID ? $attempted_user->ID : null,
+			'$user_id'      => (string) $attempted_user->ID ?? null,
```
Comment (millerf) [inline src/inc/woocommerce-actions.php:129]: Doesn't `$attempted_user->ID ?? null` on `(string)`?
I am always confused with that...

**Review by millerf — APPROVED** [review]
code LGTM and tests are passing 👍
